docs/guides/managing_packages.md

```diff
@@ -0,0 +1,122 @@
+During the development of agents and services, users need to manage other packages, either developing them themselves or fetching already available ones. This guide will show you which are the relevant directories during development and how to get, use and publish packages.
+
+The first thing you need to know is that there are different types of packages: services, agents, connections, contracts, protocols and skills. Services are composed of agents, and agents are composed of connections, contracts, protocols and skills. All these packages can live in different places:
+
+- Inside an agent if they are being used by that agent, with the exception of services which never live inside an agent.
+- In the local registry (packages folder), a local directory that stores packages classified by vendor.
+- In the remote registry, a remote machine that serves packages over IPFS or HTTP.
+
+You can see that this setup is not that different from the one git uses.
+
+<figure markdown>
+![](../images/package_management.svg){ width="85%" height="85%" style="display: block; margin: 0 auto" }
+<figcaption>Overview of the package managing flows with the Open Autonomy framework</figcaption>
+</figure>
+
+Packages should be developed independently from any agent, inside the local registry (packages folder) and later on they can be added to an agent. **This is the recommended method as it improves separation of concerns and offers the cleanest developer experience.**
+
+## Creating your agents and services
+
+The first thing a developer needs to do before they start writing code is setting up the project. Instead of starting from scratch, we recommend to use our [developer template](https://github.com/valory-xyz/dev-template). Once setup, it will generate a virtual environment with {{open_autonomy}} installed, an empty local registry, some useful tools for checking packages and dummy tests. After this, in the same fashion git asks to initialize its configuration using `git config`, {{open_autonomy}} asks the user to set the user name, whether we are using a local or remote registry as well as the type of registry (IPFS by default). For example, initialize your registry running the following command:
+
+```bash
+autonomy init --reset --author john_doe --remote --ipfs --ipfs-node "/dns/registry.autonolas.tech/tcp/443/https"
+```
+
+Now we are in the position of creating our first agent:
+
+```bash
+autonomy create my_agent
+```
+
+This agent will have some dependencies: some packages might be already available in the remote registry and some will need to be developed. Let's say, for example, that it will need the ability to communicate using the ACN protocol. Since that package is already available on the Autonolas IPFS registry, we'll use the `add command`:
+
+```bash
+cd my_agent/
+# Remote flag is not needed here as we initialized the default registry to remote
+autonomy add protocol valory/acn:0.1.0:bafybeidetrrkvdgveu4ph5g6v53lbh7ardfspbkpstmjxctx647bzyosyy
```

Comment:
I'd add a new page which just lists all the packages and their hashes and then link to it from here. We will eventually have a more elegant solution but this will do as a way for people to see the list of available packages from Valory & open-aea

Reply:
Done